### ISSUE TITLE
twisted.test.test_sslverify.SelectVerifyImplementationTests.test_dependencyMissingWarning fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7355]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#7355|
|Type|release blocker: regression|
|Created|2014-05-15 12:20:19Z|
|Branch|https://github.com/twisted/twisted/tree/servid-error-7355|

The formatting of import errors changed at some point.  It used to be like:

```
No module named service_identity
```

On Python 3.3 it seems to be like:

```
import of 'service_identity' halted; None in sys.modules
```

The test is not equipped to handle this difference.

Attachments:

* [7355-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a1f/a1f05288f185a49dd64885b980e719ea55d13839/55ddb72f026020a1caef6e42af6f656822027b4c.diff) (2730 bytes) - added by adiroiban on 2014-05-16 14:08:30Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7355 7355
type__release_blocker__regression release blocker: regression
reporter__exarkun exarkun
priority__normal normal
milestone__None None
branch__branches_servid_error_7355 branches/servid-error-7355
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1400156419000000 1400156419000000
changetime__1445217216921098 1445217216921098
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @adiroiban<br>@adiroiban set status to `assigned`|
|-|-|

I will try to fix it now.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban removed owner<br>@adiroiban set status to `new`|
|-|-|

I have fixed this with a separate value for python 3.

I moved the checks outside of ```with SetAsideModule("service_identity")``` to make it easier to see which is the ACT code and which CHECK code.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

[#7535](https://github.com/twisted/twisted/issues/7535) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

(In [43022]) Branching to servid-error-7355.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

I have run this on the buildbots, but the 3.3 buildbot doesn't have OpenSSL and skips the tests. My local machine passes the test, but I'll wait for the buildbot to say so before merging.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [43060]) Remove the duplication in the test for the warning message when service_identity is missing.

refs [#7355](https://github.com/twisted/twisted/issues/7355)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [43061]) Python 2.6 compatibility

refs [#7355](https://github.com/twisted/twisted/issues/7355)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

> I have run this on the buildbots, but the 3.3 buildbot doesn't have OpenSSL and skips the tests. My local machine passes the test, but I'll wait for the buildbot to say so before merging.

Probably necessary to draw more attention to the buildbot issue in order to get it resolved in a timely manner.

I posted https://twistedmatrix.com/pipermail/twisted-python/2014-September/028761.html to try to move this along.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

Hi exarkun, thanks for following up, and the cleanups (which I've checked over and +1, in case it needs it).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_12"></a>|@glyph set owner to @glyph|
|-|-|

Sorry this has taken so long to address; I've been dealing with a variety of buildbot issues and getting to the point where I could fix this took a little while.

That said, the buildbots are now fixed, I forced a build, and

L

G

T

M

Since I guess adi is half an author here I'll go ahead and land myself.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_13"></a>|@glyph set status to `closed`|
|-|-|

(In [43180]) Merge servid-error-7355: Brief description

Author: exarkun, adiroiban

Reviewer: glyph

Fixes: [#7355](https://github.com/twisted/twisted/issues/7355)

Fix the Python 3 buildbots.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_14"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone deleted
